### PR TITLE
Add lisk validator - Closes #1139

### LINKED ISSUE #1139: Add lisk-validator library to be used across products and libraries
### Description

Across products and libraries, we are intensively using multiple validation libraries with custom formats.

We should create a validation library to wrap all the use-cases for custom format and use consistently.

### PR DESCRIPTION
**Pull request by shuse2**
### Description
Creating basic validator library to be used in multiple packages.

### Review checklist

* The PR resolves #1139
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-validator/package.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	"version": "0.1.0",
+	"description": "Validation library according to the Lisk protocol",
+	"author": "Lisk Foundation <admin@lisk.io>, lightcurve GmbH <admin@lightcurve.io>",
+	"license": "GPL-3.0",
+	"keywords": [
+		"lisk",
+		"blockchain",
+		"validation"
+	],
+	"homepage": "https://github.com/LiskHQ/lisk-elements/tree/master/packages/lisk-transactions#readme",
```
Comment (ishantiw) [inline packages/lisk-validator/package.json]: Change to `lisk-validator#readme`? Also, missing readme file for this repo.

File: packages/lisk-validator/package.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		"node": ">=8.10 <=10",
+		"npm": ">=5"
+	},
+	"main": "dist-node/index.js",
+	"scripts": {
+		"prestart": "./scripts/prestart.sh",
+		"start": "./scripts/start.sh",
+		"transpile": "tsc",
+		"transpile:browsertest": "tsc -p tsconfig.browsertest.json",
+		"browserify": "browserify ./dist-node/index.js -o ./dist-browser/index.js -s liskTransactions",
```
Comment (ishantiw) [inline packages/lisk-validator/package.json]: `liskValidator`

File: packages/lisk-validator/package.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		"browserify:browsertest": "browserify ./browsertest.build/test/*.js -o ./browsertest.build/browsertest.js -s liskTransactions",
```
Comment (ishantiw) [inline packages/lisk-validator/package.json]: `liskValidator`

File: packages/lisk-validator/package.json [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		"uglify:browsertest": "uglifyjs -o ./browsertest.build/browsertest.min.js ./browsertest.build/browsertest.js",
+		"clean": "./scripts/clean.sh",
+		"format": "prettier --write \"*.{ts,js,json}\" \"{src,test}/**/*.{ts,js,json}\"",
+		"lint": "tslint --format verbose --project .",
+		"lint:fix": "npm run lint -- --fix",
+		"test": "TS_NODE_PROJECT=./test/tsconfig.json nyc mocha test/{,/**/,/**/**/}/*.ts",
+		"test:watch": "npm test -- --watch",
+		"test:watch:min": "npm run test:watch -- --reporter=min",
+		"test:node": "npm run build:check",
+		"serve:start": "http-server -p 11545 ./browsertest &",
```
Comment (ishantiw) [inline packages/lisk-validator/package.json]: `11545` is used by transactions, maybe use `11547`

File: packages/lisk-validator/package.json [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		"serve:stop": "kill $(lsof -t -i:11545) || true",
```
Comment (ishantiw) [inline packages/lisk-validator/package.json]: `11547`

File: packages/lisk-validator/package.json [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+		"pretest:browser": "npm run serve:stop && npm run build:browsertest && npm run serve:start",
+		"test:browser": "wait-on http://localhost:11545 && cypress run --config baseUrl=http://localhost:11545 --env ROOT_DIR=\"${PWD##*/}\"",
```
Comment (ishantiw) [inline packages/lisk-validator/package.json]: `11547`

**Review by ishantiw — DISMISSED** [review]
Looks good, just a few comments on `package.json`.  Browsertest fails on Jenkins but passes the whole test but it fails locally with error due to same ports.

File: packages/lisk-validator/src/validation.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+		return false;
+	}
+
+	return validator.isInt(num);
+};
+
+export const isValidInteger = (num: unknown): boolean =>
+	typeof num === 'number' ? Math.floor(num) === num : false;
+
+export const isUnique = (values: ReadonlyArray<string>): boolean => {
```
Comment (pablitovicente) [inline packages/lisk-validator/src/validation.ts]: Maybe rename this to `hasNoDuplicates`? isUnique seems to imply a value is being inspected to be unique in a set

**Review by pablitovicente — DISMISSED** [review]
Other than ¡shan comments looks good. I also made a comment about a name of one of the methods but is minor so I'm approving.

**Review by ishantiw — DISMISSED** [review]

**Review by pablitovicente — DISMISSED** [review]

**Review by pablitovicente — APPROVED** [review]

**Pull request merged**